app/models/app_resource_monitor/hud_references_inspector.rb

```diff
@@ -0,0 +1,165 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# report on invalid HUD references
+class AppResourceMonitor::HudReferencesInspector
+  def self.enrollment_references
+    new.enrollment_references
+  end
+
+  def self.project_references
+    new.project_references
+  end
+
+  def self.client_references
+    new.client_references
+  end
+
+  def self.duplicate_ids
+    new.duplicate_ids
+  end
+
+  def duplicate_ids
+    [
+      Hmis::Hud::Affiliation,
+      Hmis::Hud::Assessment,
+      Hmis::Hud::CeParticipation,
+      Hmis::Hud::Client,
+      Hmis::Hud::CurrentLivingSituation,
+      Hmis::Hud::CustomAssessment,
+      Hmis::Hud::CustomCaseNote,
+      Hmis::Hud::CustomClientAddress,
+      Hmis::Hud::CustomClientContactPoint,
+      Hmis::Hud::CustomClientName,
+      Hmis::Hud::CustomService,
+      Hmis::Hud::Disability,
+      Hmis::Hud::EmploymentEducation,
+      Hmis::Hud::Enrollment,
+      Hmis::Hud::Event,
+      Hmis::Hud::Exit,
+      Hmis::Hud::Funder,
+      Hmis::Hud::HealthAndDv,
+      Hmis::Hud::HmisParticipation,
+      Hmis::Hud::IncomeBenefit,
+      Hmis::Hud::Inventory,
+      Hmis::Hud::ProjectCoc,
+      Hmis::Hud::Service,
+      Hmis::Hud::YouthEducationStatus,
+    ].flat_map do |model|
+      arel_table = model.arel_table
+      hud_key = arel_table[model.hud_key]
+      data_sources.map do |data_source|
+        scope = model.where(data_source: data_source)
+        {
+          data_source_id: data_source.id,
+          table_name: model.table_name,
+          duplicates: scope.where(data_source: data_source).select(hud_key).group(hud_key).having('count(*) > 1').count.values.sum,
+        }
+      end
+    end
+  end
+
+  def client_references
+    # models that reference a client via (PersonalID)
+    [
+      Hmis::Hud::Assessment,
+      Hmis::Hud::CurrentLivingSituation,
+      Hmis::Hud::CustomAssessment,
+      Hmis::Hud::CustomCaseNote,
+      Hmis::Hud::CustomClientAddress,
+      Hmis::Hud::CustomClientContactPoint,
+      Hmis::Hud::CustomClientName,
+      Hmis::Hud::CustomService,
+      Hmis::Hud::Disability,
+      Hmis::Hud::EmploymentEducation,
+      Hmis::Hud::Enrollment,
+      Hmis::Hud::Event,
+      Hmis::Hud::Exit,
+      Hmis::Hud::HealthAndDv,
+      Hmis::Hud::IncomeBenefit,
+      Hmis::Hud::Service,
+      Hmis::Hud::YouthEducationStatus,
+    ].flat_map do |model|
```

Comment:
I swear we have this somewhere as well but I can't find it.  Something like:
```
GrdaWarehouse::Hud.models.by_hud_filename.values.select { |model| model.column_names.include?('PersonalID') }
```
as a starting point would feel less brittle to me.